diff --git a/Android.bp b/Android.bp
index 3842c905..31e1953e 100644
--- a/Android.bp
+++ b/Android.bp
@@ -36,6 +36,7 @@ rust_binary {
     name: "crosvm",
     defaults: ["crosvm_defaults"],
     host_supported: true,
+    prefer_rlib: true,
     crate_name: "crosvm",
     cargo_env_compat: true,
     srcs: ["src/main.rs"],
@@ -93,7 +94,14 @@ rust_binary {
                 "gpu",
             ],
         },
+        linux_bionic_arm64: {
+            relative_install_path: "aarch64-linux-bionic",
+        },
+        darwin: {
+            enabled: false,
+        },
         linux_glibc_x86_64: {
+            relative_install_path: "x86_64-linux-gnu",
             features: [
                 "gdb",
                 "gdbstub",
