diff --git a/devices/Android.bp b/devices/Android.bp
index 40ee2404..097f777d 100644
--- a/devices/Android.bp
+++ b/devices/Android.bp
@@ -26,6 +26,8 @@ rust_test {
     edition: "2018",
     features: [
         "audio",
+        "gfxstream",
+        "gpu",
         "usb",
     ],
     rustlibs: [
@@ -41,6 +43,7 @@ rust_test {
         "libfuse_rust",
         "libfutures",
         "libgetopts",
+        "libgpu_display",
         "libhypervisor",
         "libkvm_sys",
         "liblibc",
@@ -52,6 +55,7 @@ rust_test {
         "libp9",
         "libpower_monitor",
         "libresources",
+        "librutabaga_gfx",
         "libserde",
         "libserde_json",
         "libsmallvec",
@@ -71,6 +75,19 @@ rust_test {
         "libenumn",
         "libremain",
     ],
+    target: {
+        host: {
+            shared_libs: ["libvirglrenderer"],
+        },
+        android: {
+            shared_libs: ["libdrm"],
+            static_libs: [
+                "libepoxy",
+                "libgbm",
+                "libvirglrenderer",
+            ],
+        },
+    },
     shared_libs: ["libgfxstream_backend"],
 }
 
@@ -85,6 +102,8 @@ rust_library {
     edition: "2018",
     features: [
         "audio",
+        "gfxstream",
+        "gpu",
         "usb",
     ],
     rustlibs: [
@@ -100,6 +119,7 @@ rust_library {
         "libfuse_rust",
         "libfutures",
         "libgetopts",
+        "libgpu_display",
         "libhypervisor",
         "libkvm_sys",
         "liblibc",
@@ -111,6 +131,7 @@ rust_library {
         "libp9",
         "libpower_monitor",
         "libresources",
+        "librutabaga_gfx",
         "libserde",
         "libserde_json",
         "libsmallvec",
